x/tokenfactory/types/params.go

```diff
@@ -1,30 +1,87 @@
 package types
 
 import (
+	fmt "fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/neutron-org/neutron/app/params"
+)
+
+// Parameter store keys.
+var (
+	KeyDenomCreationFee        = []byte("DenomCreationFee")
+	DefaultNeutronDenom        = params.DefaultDenom
+	KeyFeeCollectorAddress     = []byte("FeeCollectorAddress")
+	DefaultFeeCollectorAddress = ""
 )
 
 // ParamTable for gamm module.
 func ParamKeyTable() paramtypes.KeyTable {
 	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
 }
 
-func NewParams(_ sdk.Coins) Params {
-	return Params{}
+func NewParams(denomCreationFee sdk.Coins, feeCollectorAddress string) Params {
+	return Params{
+		DenomCreationFee:    denomCreationFee,
+		FeeCollectorAddress: feeCollectorAddress,
+	}
 }
 
 // default gamm module parameters.
 func DefaultParams() Params {
-	return Params{}
+	return Params{
+		DenomCreationFee:    sdk.NewCoins(sdk.NewInt64Coin(DefaultNeutronDenom, 10_000_000)),
+		FeeCollectorAddress: DefaultFeeCollectorAddress,
+	}
 }
 
 // validate params.
 func (p Params) Validate() error {
-	return nil
+	if err := validateDenomCreationFee(p.DenomCreationFee); err != nil {
+		return err
+	}
+
+	return validateFeeCollectorAddress(p.FeeCollectorAddress)
 }
 
 // Implements params.ParamSet.
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{}
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyDenomCreationFee, &p.DenomCreationFee, validateDenomCreationFee),
+		paramtypes.NewParamSetPair(KeyFeeCollectorAddress, &p.FeeCollectorAddress, validateFeeCollectorAddress),
+	}
+}
+
+func validateDenomCreationFee(i interface{}) error {
+	v, ok := i.(sdk.Coins)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v.Validate() != nil {
+		return fmt.Errorf("invalid denom creation fee: %+v", i)
+	}
+
+	return nil
+}
+
+func validateFeeCollectorAddress(i interface{}) error {
+	v, ok := i.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	// Reserve might be explicitly empty in test environments
```

Comment:
**Reserve** might be -> **Treasury** might be? also, I think we should remove all the reserve/treasury mentions in this PR: it's not treasury or reserve, it's fee collector